### PR TITLE
Complete Sprint 6 Prep Task 8: Convexity Test Fixtures

### PR DESCRIPTION
**Pull request by jeffreyhorn**
Create comprehensive test fixture library for convexity detection with 13 minimal GAMS NLP models covering convex, non-convex, and edge cases.

Task 8: Create Convexity Test Fixtures
- Created 5 convex models (should NOT warn):
  * linear_program.gms - Simple LP
  * convex_qp.gms - Quadratic program (x² + y²)
  * convex_exponential.gms - Exponential objective
  * convex_log_barrier.gms - Log barrier function
  * convex_inequality.gms - Convex constraints

- Created 5 non-convex models (should warn):
  * nonconvex_circle.gms - Circle equality (x² + y² = 4)
  * nonconvex_trig_eq.gms - Trig equality (sin + cos = 0)
  * nonconvex_bilinear.gms - Bilinear term (x*y)
  * nonconvex_quotient.gms - Division (x/y)
  * nonconvex_odd_power.gms - Cubic term (x³)

- Created 3 edge case models (ambiguous):
  * mixed_convex_nonconvex.gms - Convex obj, nonlinear equality
  * convex_with_trig_ineq.gms - Trig inequality
  * nearly_affine.gms - Almost linear with x² = 4

- Created comprehensive README.md:
  * Complete test catalog with expected results
  * Usage examples for testing
  * Integration with convexity detection pipeline
  * Sprint 6 testing goals and targets

Model Features:
- All models <20 lines (minimal examples)
- Consistent structure with comment headers
- Expected classification documented in each file
- Parse-ready for current nlp2mcp parser

Sprint 6 Testing Targets:
- Recall: 100% detection of 5 non-convex models
- Precision: 0% false positives on 5 convex models
- Consistency: Predictable behavior on edge cases
- Performance: <1 second for all 13 models

Non-Convex Indicators:
1. Nonlinear equalities (any)
2. Bilinear terms (x*y)
3. Division (x/y)
4. Odd powers (x³, x⁵)
5. Certain trig patterns

Expected Warning Format:
"Warning: Non-convex problem detected
Nonlinear equality constraint may cause PATH solver to fail Action: Consider using NLP solver instead of PATH"

Files Created:
- tests/fixtures/convexity/README.md (240 lines)
- tests/fixtures/convexity/*.gms (13 models)

Updated:
- docs/planning/EPIC_2/SPRINT_6/PREP_PLAN.md
  * Marked Task 8 as COMPLETE
  * Checked off all 6 acceptance criteria
- CHANGELOG.md
  * Added comprehensive Task 8 summary

All acceptance criteria met:
✓ 5+ convex models
✓ 5+ non-convex models
✓ 3+ edge cases
✓ Each model <20 lines
✓ Expected warnings documented in README
✓ All models parse successfully

### CODE REVIEW

File: tests/fixtures/convexity/mixed_convex_nonconvex.gms [hunk 1]
```diff
@@ -0,0 +1,20 @@
+$title Mixed Convex/Non-Convex - Edge Case Test Fixture
+
+* Expected Classification: NON-CONVEX (due to nonlinear equality)
+* Expected Warning: Convex objective but nonlinear equality constraint
+* Description: Convex objective with non-convex feasible set
+
+Variables x, y, obj;
+
+Equations objdef, convex_obj_part, nonlinear_eq;
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply tests/fixtures/convexity/mixed_convex_nonconvex.gms]: fixed.
